releasenotes-builder/src/main/java/com/github/checkstyle/Main.java

```diff
@@ -167,24 +168,69 @@ private static void runPostGeneration(Multimap<String, ReleaseNotesMessage> rele
                 TemplateProcessor.getTemplateVariables(releaseNotes, releaseNumber);
 
         if (cliOptions.isGenerateAll() || cliOptions.isGenerateXdoc()) {
+            final boolean external = cliOptions.getXdocTemplate() != null;
```

Comment:
lets avoid "external" term.
There should be default template and provided by user.
I would get rid  of such temporal variable at all.

We just need to give to TemplateProcessor some stream to template, or just give it Template instance, or ... .
Main class should orchestrate and play with options. TemplateProcessor should just be configured in exact way and do generation.

Reply:
Done.